libs/ledger-live-common/src/families/ethereum/synchronisation.ts

```diff
@@ -163,11 +163,14 @@ export const getAccountShape: GetAccountShape = async (
     subOperations: inferSubOperations(o.hash, subAccounts),
   }));
   const operations = mergeOps(initialStableOperations, newOps);
+  const pendingOperations = initialAccount?.pendingOperations || [];
 
   const nfts = isNFTActive(currency)
     ? mergeNfts(
         initialAccount?.nfts || [],
-        nftsFromOperations(operations).filter((n) => n.amount.gt(0))
+        nftsFromOperations(operations.concat(pendingOperations)).filter((n) =>
```

Comment:
two problems i can see:
- we are not supposed to add pending operations during sync, it's the UI that reconciliate this, at least in the case of operations. it's the job of the whole "groupOperations" logic. we may want to build higher level helpers for the UI and try to preserve the synchronisation from these "merging" logics as much as possible. because we want to minimize the duplicates in our models, even tho indeed there is always a memory vs cpu tradeoff
- we are supposed to "dedup" the operations per id, for the topics i mentioned in previous comment.

Reply:
The only thing that couldn't really be changed is here. Removed the concat but replaced it by a `mergeOps` of operations + pending.

I can't really figure out how to do it differently without having to write tons and tons of code everywhere to do it in the UI instead, which seems very counter productive to me 🤷‍♂️